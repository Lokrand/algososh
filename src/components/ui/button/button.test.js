import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Button } from "./button";

describe("test <Button /> component", () => {
  it("Button render without text", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Button render with text", () => {
    const tree = renderer.create(<Button text="Добавить" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("when Button disabled", () => {
    const tree = renderer.create(<Button disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Button with loading", () => {
    const tree = renderer.create(<Button isLoader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Button with callback", () => {
    window.alert = jest.fn();
    render(<Button text="Добавить" onClick={alert("Добавлено")} />);
    const button = screen.getByText("Добавить");
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith("Добавлено");
  });
});
