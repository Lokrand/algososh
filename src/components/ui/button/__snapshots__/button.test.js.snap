// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Button /> component Button render with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Добавить
  </p>
</button>
`;

exports[`Test <Button /> component Button render without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Test <Button /> component Button with loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    id="string-spinner"
    src="loader.svg"
  />
</button>
`;

exports[`Test <Button /> component when Button disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
